src/libutil/Makefile.am

```diff
@@ -30,7 +30,9 @@ libutil_la_SOURCES = \
 	sha256.h \
 	macros.h \
 	aux.c \
-	aux.h
+	aux.h \
+	strlcpy.c \
+	strlcpy.h
```

Comment:
commit message for 1492474f8df8524d9c2dcead81a3d87e0c043367: since MUNGE `path.[ch]` is no longer being added in this PR, justification should be updated.

"because we likesssssssss it?"

Reply:
How about this? 
```
    libutil: add strlcpy
    
    Problem: strncpy(3) is a frequent source of programmer mistakes
    which directly result in security vulnerabilities. In order to be
    used safely, strncpy(3) requires an initialized buffer, or manual NUL
    termination, plus manual checks for length to ensure a result buffer
    without NUL termination is not created. This should not have to be
    the case in a project where the purpose is to create secure code.
    
    strlcpy(3), on the other hand, is designed "to be a safer, more
    consistent, and less error prone replacement" for strncpy(3) [1]. With
    its guarantee of NUL termination and safer semantics, it behooves us
    to include this implementation in the flux-security project.
    
    Note that since strlcpy(3) is known not be included in glibc, we can
    include it wholesale here. However, when porting to BSD systems
    presence of strlcpy() in libc may need to be detected and this
    source file excluded from the build if an implementation is found.
    
    [1] https://man.openbsd.org/strlcpy
```

Reply:
I'm also fine with switching back to `strncpy(3)` if you'd prefer. It is possible to safely use with proper length checks and initialization, and it is only used a handful of times currently. Perhaps our use of LGTM and ASan/UBSan and future use of fuzzing (if we can get that working) will make mistakes less likely.

Reply:
Wow nice commit message!  I say let's keep it.